controllers/update_status.go

```diff
@@ -479,14 +478,15 @@ func validateProcessGroups(ctx context.Context, r *FoundationDBClusterReconciler
 		}
 
 		if isBeingRemoved {
-			processGroup.MarkForRemoval()
+			if processGroup.RemovalTimestamp.IsZero() {
```

Comment:
What is this change about?

Reply:
I made this change to make sure we are not updating the removalTimestamp again. We can move this later into the `MarkForRemoval` method to have a check if the removalTimestamp is already set.

Reply:
And that timestamp is for user consumption, I guess?

Reply:
Correct. That's not a huge issue and I created a follow up to solve this directly in `MarkForRemoval`: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1611. The idea for this is to give the user an indication when this process group was marked for removal. The operator it self doesn't use the actual value and only checks if the value is set to a value != nil.